src/realm/object-store/impl/realm_coordinator.hpp

```diff
@@ -252,6 +252,8 @@ class RealmCoordinator : public std::enable_shared_from_this<RealmCoordinator> {
         REQUIRES(m_realm_mutex);
     void do_get_realm(Realm::Config config, std::shared_ptr<Realm>& realm, util::Optional<VersionID> version,
                       util::CheckedUniqueLock& realm_lock) REQUIRES(m_realm_mutex);
+    void open_and_init_realm(Realm::Config config, std::shared_ptr<Realm>& realm, util::Optional<VersionID> version,
```

Comment:
I am essentially transforming `do_get_realm` into a facade that just invokes the old logic for opening a realm, and catches possible exceptions, cleaning up the realm folder if something bad happened.